packages/workbox-webpack-plugin/src/index.js

```diff
@@ -0,0 +1,108 @@
+const path = require('path');
+const getEntries = require('./lib/get-webpack-manifest-entries');
+const generateManifest = require('./lib/generate-webpack-manifest');
+const generateSW = require('./lib/generate-webpack-sw');
+const webpackAsset = require('./lib/utils/webpack-asset');
+const copyWorkboxSW = require('./lib/utils/copy-workbox-sw');
+const {setReadFile} = require('./lib/utils/read-file');
+
+/**
+ * Use the instance of this in the plugins array of the webpack config.
+ *
+ * @example
+ * const WorkboxBuildWebpackPlugin = require('workbox-webpack-plugin');
+ * .
+ * module.exports = {
+ *   ...
+ *   plugins: [
+ *    new WorkboxBuildWebpackPlugin({
+ *      chunks: ['main'],
+ *      excludeChunks: ['mutable'],
+ *      manifestFilename: string,
+ *      manifestVarName: '__file_manifest',
+ *      swSrc: './src/sw.js',
+ *     });
+ *   ]
+ * };
+ *
+ * @module workbox-webpack-plugin
+ */
+class WorkboxWebpackPlugin {
+  /**
+   * Creates an instance of WorkboxBuildWebpackPlugin.
+   *
+   * @param {Object} [config] All the options as passed to `workbox-build`.
+   */
+  constructor(config) {
+    this._config = config || {};
+  }
+
+  /**
+   * @private
+   * @return {Object} The configuration for a given compilation.
+   */
+  get config() {
```

Comment:
Why are you following this model of exporting an object that seems to read and write to the config option?

Reply:
I had a similar question. I thought that this might be trying to replicate a pattern used in the sw-precache webpack plugin, but I don't think this model in this form was used there (I could be wrong) :)

Reply:
No need for this! Will completely change this "config" when I clean up the api

Reply:
👍